File: Agilité/Verrocchio/Splinter.java

/**
 * Décrivez votre classe Splinter ici.
 *
 * @author (votre nom)
 * @version (un numéro de version ou une date)
 */
public class Splinter
{
    // variables d'instance - remplacez l'exemple qui suit par le vôtre
    private String nom;
    private int anneeNaissance;
    private Eleve maitreDe;
    private Verrochio atelier;
    /**
     * Constructeur d'objets de classe Splinter
     */
    public Splinter(String nom, int anneeNaissance) {
        this.nom = nom;
        this.anneeNaissance = anneeNaissance;
    }
    /**
     * getter and setter
    **/
    public String getNom() {
        return this.nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }


    /**
     * Fonction 
       **/
    public void prendreElev(Eleve artiste) {
        this.atelier.ajouterArtiste(artiste);
    }

    public void enCours() {
        this.maitreDe.apprentissage();
    }


}

File: Splinter.py
import dataclasses

import Eleve
import Music
import Verrocchio


class Splinter:
    def __init__(self, nom, anneeNaissance):
        self.nom = nom
        self.anneeNaissance = anneeNaissance
        self.maitreDe = Eleve
        self.atelier = Verrocchio
        self.music = Music

    def getNom(self):
        return self.nom

    def setNom(self, nom):
        self.nom = nom

    def ajoutEleve(self, artiste):
        self.atelier.ajouter_artiste(artiste)

    def enCours(self):
        self.maitreDe.apprentissage()

    def ordonne_couleur(self, artiste):
        self.maitreDe.prepare_couleur(artiste)

# ---------------------------------------------------------------------- Partie music
    def fait_decouvrir_musique(self, music):
        self.atelier.ecouter_musique(music.getNom())

def getNom(nom):
    return nom


def ajoutEleve(artiste, atelier=Verrocchio):
    return atelier.ajouter_artiste(artiste)